robot/client/client.go

```diff
@@ -537,6 +537,12 @@ func (rc *RobotClient) connectWithLock(ctx context.Context) error {
 		} else {
 			err = multierr.Combine(err, grpcErr)
 		}
+
+		// instead of returning two context.DeadlineExceeded errors, tell the user how to fix
+		if errors.Is(err, context.DeadlineExceeded) && errors.Is(grpcErr, context.DeadlineExceeded) && errors.Is(grpcErr, rpc.ErrMDNSQuery) {
+			return fmt.Errorf("failed to connect to robot within time limit. check network connection and try again. " +
+				"see http://docs.viam.com/dev/tools/common-errors/#conn-time-out for troubleshooting steps")
```

Comment:
same nit as other pr, can you pull this out into a constant, and also have `https` instead of `http`?